src/run_encoded_array.cpp

```diff
@@ -0,0 +1,73 @@
+#include "sparrow/layout/run_end_encoded_layout/run_end_encoded_array.hpp"
+#include "sparrow/layout/primitive_array.hpp"
+#include "sparrow/layout/dispatch.hpp"
+#include "sparrow/layout/dispatch_lib.hpp"
+
+
+
+namespace sparrow   
+{
+    template<class T>
+    struct is_usable_array : std::false_type
+    {
+    };
+    template<>
+    struct is_usable_array<primitive_array<std::uint16_t>> : std::true_type
+    {
+    };
+    template<>
+    struct is_usable_array<primitive_array<std::uint32_t>> : std::true_type
+    {
+    };
+    template<>
+    struct is_usable_array<primitive_array<std::uint64_t>> : std::true_type
+    {
+    };
```

Comment:
```suggestion
    template<class T>
    concept usable_array =
             mpl::is_type_instance_of_v<T, primitive_array> && (
             std::same_as<typename T::value_type, std::uint16_t> ||
             std::same_as<typename T::value_type, std::uint32_t> ||
             std::same_as<typename T::value_type, std::uint64_t>);
    };
```
or something similar

Reply:
it should be inner_value_type, but your suggestions are fine with me